lib/pharos/phases/validate_host.rb

```diff
@@ -36,10 +38,17 @@ def gather_host_facts
         @host.os_release = os_release
         @host.cpu_arch = cpu_arch
         @host.hostname = hostname
-        @host.checks = @host.role == 'master' ? master_checks : worker_checks
+        @host.checks = host_checks
         @host.private_interface_address = private_interface_address(@host.private_interface) if @host.private_interface
       end
 
+      def check_role
+        return if @host.master? && @host.checks['ca_exists'] == true
+        return if @host.worker? && @host.checks['kubelet_configured'] && @host.checks['ca_exists'] != true
```

Comment:
I think the check for `@host.checks['kubelet_configured']` here is the wrong way around - this only accepts if the kubelet is configured, which is the broken part.